silverback/types.py

```diff
@@ -47,14 +50,41 @@ class _BaseDatapoint(BaseModel):
 Int96 = Annotated[int, Field(ge=-(2**95), le=2**95 - 1)]
 # NOTE: only these types of data are implicitly converted e.g. `{"something": 1, "else": 0.001}`
 ScalarType = bool | Int96 | float | Decimal
+# NOTE: Interesting side effect is that `int` outside the INT96 range parse as `Decimal`
+#       This is okay, preferable actually, because it means we can store ints outside that range
```

Comment:
@mikeshultz this is actually kinda interesting, it will "auto-parse" values outside the INT96 range as Decimal, which can actually be (less efficiently) encoded by Parquet, avoiding raising a bug about it!

Reply:
```py
In [14]: Datapoints(root={"return_value": 1324})
Out[14]: Datapoints(root={'return_value': ScalarDatapoint(type='scalar', data=1324)})

In [15]: Datapoints(root={"return_value": 2**291})
Out[15]: Datapoints(root={'return_value': ScalarDatapoint(type='scalar', data=Decimal('3978585891278293137243057985174566720803649206378781739523711815145275976100267004264448'))})
```